pyannote/database/util.py

```diff
@@ -382,6 +382,25 @@ def load_uem(file_uem):
 
     return timelines
 
+def load_lab(path):
+    """Load LAB file
+
+    Parameter
+    ---------
+    file_lab : `str`
+        Path to LAB file.
+
+    Returns
+    -------
+    data : `pd.DataFrame`
+    """
+
+    names = ["start", "end", "label"]
+    dtype = {"start": float, "end": float, "label": str}
+    data = pd.read_csv(path, names=names, dtype=dtype, delim_whitespace=True)
```

Comment:
I'd rather have `load_lab` return an `Annotation` instance, for consistency with `load_rttm`.

Since LAB files do not contain a `uri` field, you should probably updated `load_lab` signature to something like:

```python
def load_lab(path, uri: str = None) -> Annotation:
```

Reply:
It makes sense!
If the uri is not provided the function extracts it from path in this way:
`if uri is None:
        uri = path.split("/")[-1].split(".")[0]`

I'm a newbie in python, is there a better way to define this process?

Reply:
`
def load_lab(path, uri: str = None) -> Annotation:
    # If uri is not given the function extracts it from path
    if uri is None:
        uri = path.split("/")[-1].split(".")[0]

    names = ["start", "end", "label"]
    dtype = {"start": float, "end": float, "label": str}
    data = pd.read_csv(path, names=names, dtype=dtype, delim_whitespace=True)

    annotations = dict()
    annotation = Annotation(uri=uri)
    for i, turn in data.iterrows():
        segment = Segment(turn.start, turn.end)
        annotation[segment, i] = turn.label
    annotations[uri] = annotation

    return annotations
`
This is my new version of the load_lab.py file

Reply:
Since a `lab` file can only contains annotations for a single audio file, I suggested to return an `Annotation`, not a dictionary of `Annotation`.

Regarding missing `uri`, there is no need to "guess" it from the path. Simply use None if that is the case.

Reply:
Ok this is the result:
```
def load_lab(path, uri: str = None) -> Annotation:
    names = ["start", "end", "label"]
    dtype = {"start": float, "end": float, "label": str}
    data = pd.read_csv(path, names=names, dtype=dtype, delim_whitespace=True)

    annotation = Annotation(uri=uri)
    for i, turn in data.iterrows():
        segment = Segment(turn.start, turn.end)
        annotation[segment, i] = turn.label

    return annotation

class LabLoader:
    """LAB loader

    Parameters
    ----------
    path : str
        Path to LAB file with ProtocolFile
        (e.g. "/path/to/{uri}.lab")
    """

    def __init__(self, path: Text = None):
        super().__init__()

        self.path = str(path)
        
        _, placeholders, _, _ = zip(*string.Formatter().parse(self.path))
        self.placeholders_ = set(placeholders) - set([None])
        self.loaded_ = dict()
        
    def __call__(self, file: ProtocolFile) -> Annotation:

        uri = file["uri"]

        if uri in self.loaded_:
            return self.loaded_[uri]

        sub_file = {key: file[key] for key in self.placeholders_}
        loaded = load_lab(self.path.format(**sub_file), uri)
        
        # do not cache timelines when there is one LAB file per "uri"
        # since loading it should be quite fast
        if "uri" in self.placeholders_:
            return loaded

        # when there is more than one file in loaded LAB, cache them all
        # so that loading future "uri" will be instantaneous
        self.loaded_[uri] = loaded

        return self.loaded_[uri]
```
In the init I don't load lab files because I don't know the uri, as you suggested me.